README.md

```diff
@@ -19,10 +19,93 @@ build and deployment processes.
 
 [**Documentation is Available Online!**][3]
 
+# Connection Pool
+
+The NuoDB Node.js driver comes with a built in connection pool available.
+
+### Arguments
+
+**connectionConfig:** the configuration that will be used to create the connection in the pool, required argument.
+
+**minAvailable:** initial size of the pool and the number of connection the pool will aim to maintain, default of 10 is used if no argument is provided.
+
+**maxAge:** Amount of time from connection creation until it will age out, default of 300000ms (5 minutes) is used if no argument is provided.
+
+**checkTime:** how often the pool will run an internal liveliness check on free connections, default of 120000ms(2 minutes) is used if no argument is provided.
+
+**maxLimit:** hard cap on the amount of live connection the pool can maintain, default of 200 is used if no argument is provided. If 0, the pool will have no hard cap.
+
+**connectionRetryLimit:** amount of times a pool will attempt to create a connection, default of 5 is used if no argument is provided.
+
+**id:** optional argument to give the pool an id. As default the pool will be provided the “new Date().getTime()” at the time of its creation as its id.
+
+arguments should be provided to the pool as an object as such:
+
+### Usage
+
+```
+const myPool = new Pool({
+    minAvailalbe: <arg>,
+    connectionConfig: <connection config obj>,
+    maxAge: <arg>,
+    checkTime: <arg>,
+    maxLimit: <arg>,
+    connectionRetryLimit: <arg>,
+    id: <arg>
+})
+```
+
+### Methods
+
+After a pool is created the user must initiate it by using the init method:
+
+```
+await Pool.init()
+```
+
+This will populate the pool. The pool is unaivable until this completes.
+
+Once the pool has been successfully initialized it is ready for use. The user uses:
+
+```
+const newConnection = await Pool.requestConnection()
+```
+
+to get a connection, this method will return a connection ready for use.
+
+Once the user has finished using a connection, it can be returned to the pool with:
+
+```
+await Pool.releaseConnection(<connection>)
+```
+
+Connections that have been released back to the connection should not be used anymore.
+Connections that are somehow failing should be returned to the pool where they will dealt with and replaced.
+
+A pool can be shutdown with:
+
+```
+await Pool.closePool()
+```
+
+This will close all of the pools connections regardless of weither or not they are in use.
+
+Users cannot manually close connections provided by the pool, and calling:
+
+```
+connection.close()
+```
+
+on a connection provided by the pool will have the same effect as calling:
+
+```
+Pool.releaseConnection(connection)
+```
```

Comment:
This would probably be more readable with code spans (use a single `) so that is reads

Users cannot manually close connections provided by the pool, and calling `connection.close()` on a connection provided by the pool will have the same effect as calling `Pool.releaseConnection(connection)`.

Reply:
Good call, made the change